src/schema/posts.ts

```diff
@@ -859,6 +925,80 @@ export const resolvers: IResolvers<any, Context> = {
 
       return { _: true };
     },
+    editPost: async (
+      source,
+      args: EditPostArgs,
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      const { id, image } = args;
+      const { con, userId } = ctx;
+      const title = args.title?.trim() ?? '';
+      const content = args.content?.trim() ?? '';
+
+      await con.transaction(async (manager) => {
+        const repo = manager.getRepository(Post);
+        const post = (await repo.findOneByOrFail({ id })) as
+          | WelcomePost
+          | FreeformPost;
```

Comment:
Does this not force cast it to these types?

Reply:
We fetch through `Post` entity, so it can be either of the 4, this is so we can fetch both types where we allow the editting.